src/block/BaseSign.php

```diff
@@ -94,6 +104,33 @@ public function place(BlockTransaction $tx, Item $item, Block $blockReplace, Blo
 		return parent::place($tx, $item, $blockReplace, $blockClicked, $face, $clickVector, $player);
 	}
 
+	public function onInteract(Item $item, int $face, Vector3 $clickVector, ?Player $player = null) : bool{
+		if($player !== null and $item instanceof Dye){
+			$dyeColor = $item->getColor();
+			$oldColor = $this->textColor;
+
+			if($dyeColor->getRgbValue()->toARGB() === $oldColor->toARGB()){
+				return false;
+			}
+			$this->textColor = $dyeColor->getRgbValue();
```

Comment:
A special case is needed here for `BLACK`: you need to use `new Color(0,0,0)`, so that the text appears with the appropriate white outline when glowing. Using the regular dye colour will make it appear like this: 

![image](https://user-images.githubusercontent.com/14214667/148073081-0537b480-cd29-469e-ba79-50641c7a981f.png)

but it should look like this:
![image](https://user-images.githubusercontent.com/14214667/148073236-37e337fe-3b44-44bd-ba1c-2ce63eddcfac.png)